From 51efa3f9985244c5d40e33709de1022e68cca693 Mon Sep 17 00:00:00 2001
From: Victoria Martinez de la Cruz <victoria@redhat.com>
Date: Tue, 12 Jan 2016 16:43:27 -0300
Subject: [PATCH] Adds fix for mysql/mariadb configuration paths

Fix MariaDB paths (RHBZ#1288590)
Restores upstream  MYSQL_BIN_CANDIDATES (RHBZ#1293622)
---
 trove/guestagent/datastore/mysql_common/service.py | 10 ++++++----
 trove/templates/mariadb/config.template            |  2 +-
 trove/templates/mysql/config.template              |  2 +-
 3 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/trove/guestagent/datastore/mysql_common/service.py b/trove/guestagent/datastore/mysql_common/service.py
index 5e24dfc..c9fac35 100644
--- a/trove/guestagent/datastore/mysql_common/service.py
+++ b/trove/guestagent/datastore/mysql_common/service.py
@@ -69,11 +69,13 @@ OS_NAME = operating_system.get_os()
 MYSQL_CONFIG = {operating_system.REDHAT: "/etc/my.cnf",
                 operating_system.DEBIAN: "/etc/mysql/my.cnf",
                 operating_system.SUSE: "/etc/my.cnf"}[OS_NAME]
-MYSQL_SERVICE_CANDIDATES = ["mysql", "mysqld", "mysql-server"]
+MYSQL_SERVICE_CANDIDATES = ["mariadb", "mysql", "mysqld", "mysql-server"]
 MYSQL_BIN_CANDIDATES = ["/usr/sbin/mysqld", "/usr/libexec/mysqld"]
 MYSQL_OWNER = 'mysql'
 CNF_EXT = 'cnf'
-CNF_INCLUDE_DIR = '/etc/mysql/conf.d/'
+CNF_INCLUDE_DIR = {operating_system.REDHAT: "/etc/my.cnf.d/",
+                   operating_system.DEBIAN: "/etc/mysql/conf.d/",
+                   operating_system.SUSE: "/etc/my.cnf.d/"}[OS_NAME]
 CNF_MASTER = 'master-replication'
 CNF_SLAVE = 'slave-replication'
 
@@ -700,7 +702,7 @@ class BaseMySqlApp(object):
         """Clear old configs, which can be incompatible with new version."""
         LOG.debug("Clearing old MySQL config.")
         random_uuid = str(uuid.uuid4())
-        configs = ["/etc/my.cnf", "/etc/mysql/conf.d", "/etc/mysql/my.cnf"]
+        configs = ["/etc/my.cnf", "/etc/my.cnf.d"]
         for config in configs:
             try:
                 old_conf_backup = "%s_%s" % (config, random_uuid)
@@ -711,7 +713,7 @@ class BaseMySqlApp(object):
                 pass
 
     def _create_mysql_confd_dir(self):
-        conf_dir = "/etc/mysql/conf.d"
+        conf_dir = "/etc/my.cnf.d"
         LOG.debug("Creating %s." % conf_dir)
         operating_system.create_directory(conf_dir, as_root=True)
 
diff --git a/trove/templates/mariadb/config.template b/trove/templates/mariadb/config.template
index f0155eb..448af55 100644
--- a/trove/templates/mariadb/config.template
+++ b/trove/templates/mariadb/config.template
@@ -54,4 +54,4 @@ max_allowed_packet = 16M
 [isamchk]
 key_buffer = 16M
 
-!includedir /etc/mysql/conf.d/
+!includedir /etc/my.cnf.d/
diff --git a/trove/templates/mysql/config.template b/trove/templates/mysql/config.template
index 78ed025..baa6db8 100644
--- a/trove/templates/mysql/config.template
+++ b/trove/templates/mysql/config.template
@@ -53,4 +53,4 @@ max_allowed_packet = 16M
 [isamchk]
 key_buffer = 16M
 
-!includedir /etc/mysql/conf.d/
+!includedir /etc/my.cnf.d/
